jupyterlab_search_replace/search_engine.py

```diff
@@ -120,4 +120,14 @@ async def search(self, regex: str, path: str = "", max_count: int = 100):
 
             return {"matches": matches_per_files}
         else:
+            output = json.loads(output)
+            if output["type"] == "summary":
+                stats = output["data"]["stats"]
+                if (
+                    stats["matched_lines"] == 0
+                    and stats["matches"] == 0
+                    and stats["searches"] == 0
+                    and stats["searches_with_match"] == 0
+                ):
+                    return {"matches": []}
```

Comment:
We should be careful here in case the error results in invalid json

```suggestion
            try:
                output = json.loads(output)
                if output["type"] == "summary":
                    stats = output["data"]["stats"]
                    if (
                        stats["matched_lines"] == 0
                        and stats["matches"] == 0
                        and stats["searches"] == 0
                        and stats["searches_with_match"] == 0
                    ):
                        return {"matches": []}
            except (json.JSONDecodeError, KeyError):
                # If parsing the JSON fails or one key is missing
                # consider the output as invalid
                pass
```